lib/runner.js

```diff
@@ -112,7 +113,7 @@ class Runner extends EventEmitter {
 		}
 
 		if (metadata.type === 'test' && this.match.length > 0) {
-			metadata.exclusive = title !== null && matcher([title], this.match).length === 1;
+			metadata.exclusive = title === null ? this.matchTitleless : matcher([title], this.match).length === 1;
```

Comment:
I think it may be sufficient to remove the `title !== null` check. `matcher([null],["!*oo"])` gives `[null]` which passes the `length === 1` check. Perhaps use `matcher([title || ''], this.match)`, since the [matcher](https://www.npmjs.com/package/matcher) expects strings.

Reply:
I'm okay with that simplification, but you should know that it does change behavior—that would essentially treat a titleless test the same as a test with an empty string title, so `-m '*'` would truly include all tests, even the titleless ones, and there would no longer be an escape hatch to go back to the old behavior. I was trying to lean towards making this PR as conservative as possible, but I can't think of a real reason to want to distinguish between anonymous tests and empty title tests, so if you're fine with that then I'll make the change.